tests/model_registry/utils.py

```diff
@@ -0,0 +1,72 @@
+from kubernetes.dynamic import DynamicClient
+from ocp_resources.namespace import Namespace
+from ocp_resources.service import Service
+from ocp_resources.model_registry import ModelRegistry
+from kubernetes.dynamic.exceptions import ResourceNotFoundError
+
+from tests.utils import ProtocolNotSupported
+from utilities.constants import Protocols, HTTPRequest
+from tests.model_registry.constants import ModelRegistryEndpoints
+
+
+ADDRESS_ANNOTATION_PREFIX: str = "routing.opendatahub.io/external-address-"
+
+
+class TooManyServices(Exception):
+    def __init__(self, services: list[Service]):
+        self.services = services
+
+    def __str__(self) -> str:
+        return f"The Model Registry instance has too many Services, there should be only 1. List: {self.services}"
+
+
+def get_mr_service_by_label(client: DynamicClient, ns: Namespace, mr_instance: ModelRegistry) -> Service:
+    """
+    Args:
+        client (DynamicClient): OCP Client to use.
+        ns (Namespace): Namespace object where to find the Service
+        mr_instance (ModelRegistry): Model Registry instance
+
+    Returns:
+        Service: The matching Service
+
+    Raises:
+        ResourceNotFoundError: if no service is found.
+    """
+    if svc := [
+        svcs
+        for svcs in Service.get(
+            dyn_client=client,
+            namespace=ns.name,
+            label_selector=f"app={mr_instance.name},component=model-registry",
+        )
+    ]:
+        if len(svc) > 1:
+            raise TooManyServices(svc)
+        return svc[0]
```

Comment:
a suggestion, i think it is clearer to first have the positive path and then fail, i.e
```suggestion
        if len(svc) == 1:
            return svc[0]

        raise TooManyServices(svc)
```